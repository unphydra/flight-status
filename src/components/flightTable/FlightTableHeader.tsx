import { TableCell, TableHead, TableRow } from "@mui/material";
import { FlightDetail } from "../../api/flightApi";

const columns: {
  id: keyof FlightDetail;
  label: string;
}[] = [
  {
    id: "flightNumber",
    label: "Flight Number",
  },
  {
    id: "airline",
    label: "Airline",
  },
  {
    id: "origin",
    label: "Origin",
  },
  {
    id: "destination",
    label: "Destination",
  },
  {
    id: "departureTime",
    label: "Departure Time",
  },
  {
    id: "status",
    label: "Status",
  },
];

export default function FlightTableHeader() {
  return (
    <TableHead className="flight-table-header">
      <TableRow>
        {columns.map(({ label }, index) => (
          <TableCell key={index}>{label}</TableCell>
        ))}
      </TableRow>
    </TableHead>
  );
}
