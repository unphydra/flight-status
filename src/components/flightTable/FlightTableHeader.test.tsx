import FlightTableHeader from "./FlightTableHeader";

import { render, screen } from "@testing-library/react";

describe("FlightsTableHeader", () => {
  it("Should renders and match snapshot", () => {
    const { container } = render(<FlightTableHeader />);
    expect(container).toMatchSnapshot();
  });

  it("Should renders flight table header correctly", () => {
    render(<FlightTableHeader />);

    const flightNumberHeader = screen.getByText("Flight Number");
    expect(flightNumberHeader).toBeInTheDocument();

    const airlineHeader = screen.getByText("Airline");
    expect(airlineHeader).toBeInTheDocument();

    const originHeader = screen.getByText("Origin");
    expect(originHeader).toBeInTheDocument();

    const destinationHeader = screen.getByText("Destination");
    expect(destinationHeader).toBeInTheDocument();

    const departureTimeHeader = screen.getByText("Departure Time");
    expect(departureTimeHeader).toBeInTheDocument();

    const statusHeader = screen.getByText("Status");
    expect(statusHeader).toBeInTheDocument();
  });

  it("Should renders correct number of columns", () => {
    render(<FlightTableHeader />);

    const tableHead = screen.getByRole("rowgroup");
    const tableHeaders = tableHead.querySelectorAll("th");

    expect(tableHeaders.length).toBe(6);
  });
});
