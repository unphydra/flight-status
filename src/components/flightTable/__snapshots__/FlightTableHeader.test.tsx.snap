// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FlightsTableHeader > Should renders and match snapshot 1`] = `
<div>
  <thead
    class="MuiTableHead-root flight-table-header css-15wwp11-MuiTableHead-root"
  >
    <tr
      class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
    >
      <th
        class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
        scope="col"
      >
        Flight Number
      </th>
      <th
        class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
        scope="col"
      >
        Airline
      </th>
      <th
        class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
        scope="col"
      >
        Origin
      </th>
      <th
        class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
        scope="col"
      >
        Destination
      </th>
      <th
        class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
        scope="col"
      >
        Departure Time
      </th>
      <th
        class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
        scope="col"
      >
        Status
      </th>
    </tr>
  </thead>
</div>
`;
